### PR TITLE
ENH: Allow quantile to accept an int

### PR DESCRIPTION
**Pull request by dsaxton**
This change is to allow `quantile` to accept an `int` argument to make it easier to compute a set of evenly-spaced quantiles.  For example, instead of typing `s.quantile([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9])` or using `np.linspace` you can type `s.quantile(10)`.

- [ - ] closes #xxxx (no issue)
- [ x ] tests added / passed
- [ x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ x ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -305,8 +305,7 @@ Numeric
 - Bug in :meth:`Series.divmod` and :meth:`Series.rdivmod` which would raise an (incorrect) ``ValueError`` rather than return a pair of :class:`Series` objects as result (:issue:`25557`)
 - Raises a helpful exception when a non-numeric index is sent to :meth:`interpolate` with methods which require numeric index. (:issue:`21662`)
 - Bug in :meth:`~pandas.eval` when comparing floats with scalar operators, for example: ``x < -0.1`` (:issue:`25928`)
--
-
+- Enabled :meth:`quantile` to accept an integer argument.
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: this is not a valid reference is this series, dataframe, groupby what?

File: pandas/core/frame.py [hunk 2]
```diff
@@ -7840,6 +7842,12 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
         C        1 days 12:00:00
         Name: 0.5, dtype: object
         """
+        if isinstance(q, int) and q > 1:
+            return self.quantile(q=[i / q for i in range(1, q)],
```
Comment (jreback) [inline pandas/core/frame.py]: i am not sure this that intuitive 
why is it difficult to use linspace(0, 1, 10) here? would take that as an example

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 2 above)
Comment (dsaxton) [reply pandas/core/frame.py]: `linspace(0, 1, 10)` will include the min and max as well (which I think people often won't want when they're looking for quantiles) and so we actually need `linspace(0, 1, 11)`, which is a little unusual.  I was thinking that `quantile(10)` would be an easy way to get the cuts themselves for dividing the data into 10 equally-sized groups.

**Comment by mroeschke** [comment]
I'm not too fond of having the `q` parameter meaning 2 different things (the actual quantile(s) or a spacing). What's wrong with passing `np.linspace`?

**Comment by TomAugspurger** [comment]
Agreed with https://github.com/pandas-dev/pandas/pull/26350#issuecomment-491638345

**Comment by dsaxton** [comment]
Ok, I'd been using the method a fair amount and felt it might make things more convenient in some cases, but could also see it causing confusion

**Pull request closed without merging**